Gems/OpenXRVk/Code/Include/OpenXRVk/OpenXRVkDevice.h

```diff
@@ -30,9 +30,9 @@ namespace OpenXRVk
         // Create the xr specific native device object and populate the XRDeviceDescriptor with it.
         AZ::RHI::ResultCode InitDeviceInternal(AZ::RHI::XRDeviceDescriptor* instanceDescriptor) override;
         //! Get the Fov data  of the view specified by view index
-        AZ::RPI::FovData GetViewFov(AZ::u32 viewIndex) const override;
+        AZ::RHI::ResultCode GetViewFov(const AZ::u32 viewIndex, AZ::RPI::FovData& outFovData) const override;
```

Comment:
Suggest leaving the first argument as `AZ::u32 viewIndex` since the fact that `viewIndex` is unchanged internally is essentially an _implementation detail_ of `GetViewFov` and we wouldn't want the API of `GetViewFov` to change if `viewIndex` changed internally (not that it would). Generally, I think arguments passed by value should never be marked `const` for this reason (exposes/leaks an implementation detail). This comment probably applies to the other APIs introduced in this PR